crates/red_knot_python_semantic/resources/mdtest/suppressions/type-ignore.md

```diff
@@ -0,0 +1,128 @@
+# Suppressing errors with `type: ignore`
+
+Type check errors can be suppressed by a `type: ignore` comment on the same line as the violation.
+
+## Simple `type: ignore`
+
+```py
+a = 4 + test  # type: ignore
+```
+
+## Multiline ranges
+
+A diagnostic with a multiline range can be suppressed by a comment on the same line as the
+diagnostic's start or end. This is the same behavior as Mypy's.
+
+```py
+# fmt: off
+y = (
+    4 / 0  # type: ignore
+)
+
+y = (
+    4 /  # type: ignore
+    0
+)
+
+y = (
+    4 /
+    0  # type: ignore
+)
+```
+
+Pyright diverges from this behavior and instead applies a suppression if its range intersects with
+the diagnostic range. This can be problematic for nested expressions because a suppression in a
+child expression now suppresses errors in the outer expression.
+
+For example, the `type: ignore` comment in this example suppresses the error of adding `2` to
+`"test"` and adding `"other"` to the result of the cast.
+
+```py path=nested.py
+# fmt: off
+from typing import cast
+
+y = (
+    cast(int, "test" +
+            2 # type: ignore
+    )
+    + "other"  # TODO: expected-error[invalid-operator]
```

Comment:
Could we use `/ 0` here and then the test would function without a TODO?

Or maybe that muddies the comparison with pyright, which doesn't error on division by zero

Reply:
It doesn't work with `/ 0` because Red Knot doesn't support `cast` yet (and without the cast it's `Unknown / 0` which we seem to accept)